modules/network_synchronizer/scene_synchronizer.cpp

```diff
@@ -0,0 +1,1845 @@
+
+	ClassDB::bind_method(D_METHOD("register_variable", "node", "variable", "on_change_notify", "skip_rewinding"), &SceneSynchronizer::register_variable, DEFVAL(StringName()), DEFVAL(false));
+	ClassDB::bind_method(D_METHOD("unregister_variable", "node", "variable"), &SceneSynchronizer::unregister_variable);
+
+	ClassDB::bind_method(D_METHOD("get_changed_event_name", "variable"), &SceneSynchronizer::get_changed_event_name);
+
+	ClassDB::bind_method(D_METHOD("track_variable_changes", "node", "variable", "method"), &SceneSynchronizer::track_variable_changes);
+	ClassDB::bind_method(D_METHOD("untrack_variable_changes", "node", "variable", "method"), &SceneSynchronizer::untrack_variable_changes);
+
+	ClassDB::bind_method(D_METHOD("set_node_as_controlled_by", "node", "controller"), &SceneSynchronizer::set_node_as_controlled_by);
+
+	ClassDB::bind_method(D_METHOD("register_process", "node", "function"), &SceneSynchronizer::register_process);
+	ClassDB::bind_method(D_METHOD("unregister_process", "node", "function"), &SceneSynchronizer::unregister_process);
+
+	ClassDB::bind_method(D_METHOD("is_recovered"), &SceneSynchronizer::is_recovered);
+	ClassDB::bind_method(D_METHOD("is_resetted"), &SceneSynchronizer::is_resetted);
+	ClassDB::bind_method(D_METHOD("is_rewinding"), &SceneSynchronizer::is_rewinding);
+
+	ClassDB::bind_method(D_METHOD("force_state_notify"), &SceneSynchronizer::force_state_notify);
+
+	ClassDB::bind_method(D_METHOD("_on_peer_connected"), &SceneSynchronizer::_on_peer_connected);
+	ClassDB::bind_method(D_METHOD("_on_peer_disconnected"), &SceneSynchronizer::_on_peer_disconnected);
+
+	ClassDB::bind_method(D_METHOD("__clear"), &SceneSynchronizer::__clear);
+	ClassDB::bind_method(D_METHOD("_rpc_send_state"), &SceneSynchronizer::_rpc_send_state);
+	ClassDB::bind_method(D_METHOD("_rpc_notify_need_full_snapshot"), &SceneSynchronizer::_rpc_notify_need_full_snapshot);
+
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "server_notify_state_interval", PROPERTY_HINT_RANGE, "0.001,10.0,0.0001"), "set_server_notify_state_interval", "get_server_notify_state_interval");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "comparison_float_tolerance", PROPERTY_HINT_RANGE, "0.000001,0.01,0.000001"), "set_comparison_float_tolerance", "get_comparison_float_tolerance");
+}
+
+void SceneSynchronizer::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
+			if (Engine::get_singleton()->is_editor_hint())
+				return;
+
+			// TODO add a signal that allows to not check this each frame.
+			if (unlikely(peer_ptr != get_multiplayer()->get_network_peer().ptr())) {
+				reset_synchronizer_mode();
+			}
+
+			const int lowest_priority_number = INT32_MAX;
+			ERR_FAIL_COND_MSG(get_process_priority() != lowest_priority_number, "The process priority MUST not be changed, is likely there is a better way of doing what you are trying to do, if you really need it please open an issue.");
```

Comment:
Nit-picky but "it is likely there is a better" (or "it's") is more correct.

Reply:
Done ty